src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -510,11 +512,13 @@ class SchedulerActions(
   */
 object TaskStatusCollector {
   def collectTaskStatusFor(instances: Seq[Instance]): Seq[mesos.Protos.TaskStatus] = {
-    instances.flatMap { instance =>
-      instance.tasksMap.values.withFilter(task => !task.isTerminal && !task.isReserved).map { task =>
-        task.status.mesosStatus.getOrElse(initialMesosStatusFor(task, instance.agentInfo))
-      }
-    }(collection.breakOut)
+    instances.collect { case i @ Instance(_, Some(agentInfo), _, NonEmpty(tasksMap), _, _, _) => (agentInfo, tasksMap) }
+      .flatMap {
+        case (agentInfo, tasksMap) =>
+          tasksMap.values.withFilter(task => !task.isTerminal).map { task =>
```

Comment:
I am actually not sure here why was the isReserved here and whether we can remove it...

Reply:
Goof catch. I think we used to report only running instances.